libnetwork/ipam/structures.go

```diff
@@ -55,22 +55,22 @@ func (s *PoolID) String() string {
 // FromString populates the SubnetKey object reading it from string
 func (s *PoolID) FromString(str string) error {
 	if str == "" || !strings.Contains(str, "/") {
-		return types.BadRequestErrorf("invalid string form for subnetkey: %s", str)
+		return types.InvalidParameterErrorf("invalid string form for subnetkey: %s", str)
 	}
 
 	p := strings.Split(str, "/")
 	if len(p) != 3 && len(p) != 5 {
-		return types.BadRequestErrorf("invalid string form for subnetkey: %s", str)
+		return types.InvalidParameterErrorf("invalid string form for subnetkey: %s", str)
 	}
 	sub, err := netip.ParsePrefix(p[1] + "/" + p[2])
 	if err != nil {
-		return types.BadRequestErrorf("%v", err)
+		return types.InvalidParameterErrorf("%v", err)
```

Comment:
😭 so... we're converting an error to a string so that we can produce an error 😞 

I guess we could change `invalidParameter` to be a struct instead of a string;






But honestly,, at that point we should start wondering if we should just use the errdefs helpers directly;


```suggestion
		return errdefs.InvalidParameter(fmt.Errorf("invalid string form for subnetkey: %s", str))
```

That way we can also Unwrap errors etc, which currently isn't possible.

Reply:
If we want to keep the `types` helpers, we could still make the utilities a wrapper, e.g.

```go
// InvalidParameterErrorf creates an instance of InvalidParameterError
func InvalidParameterErrorf(format string, params ...interface{}) error {
	return errdefs.InvalidParameter(fmt.Errorf(format, params...))
}
```